sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/SqlBatchHandler.java

```diff
@@ -194,12 +192,22 @@ public Object invoke(Object target, Object[] args, HandleSupplier h) {
 
             private ResultIterator<?> loadChunk() {
                 // execute a single chunk and buffer
-                PreparedBatch batch = handle.prepareBatch(sql);
+                List<Object[]> currArgs = new ArrayList<>();
                 for (int i = 0; i < chunkSize && batchArgs.hasNext(); i++) {
-                    applyCustomizers(batch, batchArgs.next());
+                    Object[] batchArg = batchArgs.next();
+                    currArgs.add(Arrays.copyOf(batchArg, batchArg.length));
```

Comment:
This is to avoid keeping references to the `sharedArg` at line 307, right? If so, there is no longer a point to sharing that array - instead of copying a shared array repeatedly we can just allocate a new one for each iteration.

Reply:
No, this is to ensure that the `batchArgs` are copied before submitting to `PreparedBatch` to ensure the args aren't cleared after execute. This is the fix where `SerializableTransactionRunner` (line 55) - on a second retry will preserve these args.

Reply:
Thanks for this. Initializing sharedArg everytime we do this.